File: leo2-node/src/main/java/org/deku/leo2/node/data/repositories/CountryRepository.java
package org.deku.leo2.node.data.repositories;

import org.deku.leo2.node.data.entities.Country;
import org.deku.leo2.node.data.repositories.custom.CountryRepositoryCustom;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.data.querydsl.QueryDslPredicateExecutor;

/**
 * Created by JT on 11.05.15.
 */
public interface CountryRepository extends JpaRepository<Country, String>, QueryDslPredicateExecutor, CountryRepositoryCustom {
}

File: leoz-node/src/main/java/org/deku/leoz/node/data/repository/CountryRepository.kt
package org.deku.leoz.node.data.repository

import org.deku.leoz.node.data.jpa.MstCountry
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.data.querydsl.QuerydslPredicateExecutor
import javax.inject.Inject

/**
 * Country repository
 * Created by masc on 11.05.15.
 */
interface CountryRepository :
        JpaRepository<MstCountry, String>,
        QuerydslPredicateExecutor<MstCountry>,
        CountryRepositoryExtension


interface CountryRepositoryExtension

class CountryRepositoryImpl : CountryRepositoryExtension {
    @Inject
    private lateinit var routeRepository: CountryRepository
}
